src/web_interface/components/component_base.py

```diff
@@ -1,16 +1,41 @@
-# -*- coding: utf-8 -*-
+from types import MethodType
+from typing import Any, Callable, NamedTuple, Tuple
 
-from abc import ABCMeta, abstractmethod
+ROUTES_ATTRIBUTE = 'view_routes'
 
+GET = 'GET'
+POST = 'POST'
 
-class ComponentBase(metaclass=ABCMeta):
+
+Route = NamedTuple('Route', [('rule', str), ('methods', Tuple[str, ...])])
+
+
+class AppRoute:
+    def __init__(self, rule: str, *methods: str):
+        self.route = Route(rule, methods)
+
+    def __call__(self, view_function: Callable) -> Callable:
+        if not hasattr(view_function, ROUTES_ATTRIBUTE):
+            setattr(view_function, ROUTES_ATTRIBUTE, [])
+        getattr(view_function, ROUTES_ATTRIBUTE).append(self.route)
+        return view_function
+
+
+class ComponentBase:
     def __init__(self, app, config, api=None):
         self._app = app
         self._config = config
         self._api = api
 
         self._init_component()
 
-    @abstractmethod
     def _init_component(self):
-        pass
+        for attribute in dir(self):
+            method = getattr(self, attribute)
+            if _is_view_function(method):
+                for route in getattr(method, ROUTES_ATTRIBUTE):  # type: Route
+                    self._app.add_url_rule(rule=route.rule, view_func=method, methods=route.methods)
```

Comment:
Does this automagically set the `view_func.__name__` as target that can be used by `url_for()`?

Reply:
Yes, `url_for()` works with the endpoint name and if no custom endpoint name is used, the function name (i.e. `view_func.__name__`) is used (e.g. `url_for('show_analysis')`)